server/mailrepository/mailrepository-api/src/test/java/org/apache/james/mailrepository/MailRepositoryTest.java

```diff
@@ -0,0 +1,125 @@
+            .of(new MailAddress("rec1@domain.com"),
+                new MailAddress("rec2@domain.com"));
+        MailAddress sender = new MailAddress("sender@domain.com");
+        Mail mail = new MailImpl(name, sender, recipients, mailContent);
+        mail.setAttribute(TEST_ATTRIBUTE, "testValue");
+        return mail;
+    }
+
+    default InputStream generateMailContent() {
+        String headers = new StringJoiner("\r\n")
+            .add("Subject: test")
+            .add("Content-Type: text/plain")
+            .toString();
+        String headerBodySeparator = "\r\n\r\n";
+        String body = "original body" + RandomStringUtils.randomAlphabetic(10);
+        String end = "\r\n.\r\n";
+        return InputStreams.create(headers + headerBodySeparator + body + end, StandardCharsets.UTF_8);
+    }
+
+    default void checkMailEquality(Mail actual, Mail expected) {
+        assertAll(
+            () -> assertThat(actual.getMessage().getContent()).isEqualTo(expected.getMessage().getContent()),
+            () -> assertThat(actual.getMessageSize()).isEqualTo(expected.getMessageSize()),
+            () -> assertThat(actual.getName()).isEqualTo(expected.getName()),
+            () -> assertThat(actual.getState()).isEqualTo(expected.getState()),
+            () -> assertThat(actual.getAttribute(TEST_ATTRIBUTE)).isEqualTo(expected.getAttribute(TEST_ATTRIBUTE))
+        );
+    }
+
+    MailRepository retrieveRepository() throws Exception;
+
+    @Test
+    default void storeRegularMailShouldNotFail() throws Exception {
+        MailRepository testee = retrieveRepository();
+        Mail email = createMail("mail1");
+        testee.store(email);
+    }
+
+    @Test
+    default void retrieveShouldGetStoredMail() throws Exception {
+        MailRepository testee = retrieveRepository();
+        Mail email = createMail("mail1");
+        testee.store(email);
+        assertThat(testee.retrieve("mail1")).satisfies(actual -> checkMailEquality(actual, email));
+    }
+
+    @Test
+    default void newlyCreatedRepositoryShouldNotContainAnyMail() throws Exception {
+        MailRepository testee = retrieveRepository();
+        assertThat(testee.list()).isEmpty();
+    }
+
+    @Test
+    default void retrievingUnknownMailShouldReturnNull() throws Exception {
+        MailRepository testee = retrieveRepository();
+        assertThat(testee.retrieve("random")).isNull();
+    }
+
+    @Test
+    default void retrievingUnknownMailShouldHaveNoEffect() throws Exception {
+        MailRepository testee = retrieveRepository();
+        testee.remove("random");
+    }
+
+    @Test
+    default void listShouldReturnStoredMailsKeys() throws Exception {
+        MailRepository testee = retrieveRepository();
+        testee.store(createMail("mail1"));
+        testee.store(createMail("mail2"));
+        assertThat(testee.list()).containsExactly("mail1", "mail2");
+    }
+
+    @Test
+    @Disabled("JDBC doesn't update the message Content")
```

Comment:
Shouldn't this only be disabled for JDBC ?

Reply:
I'll remove this, it makes no sense as it doesn't even run on jdbc at this point of the history

Reply:
changed